homeassistant/components/switch/raspyrfm.py

```diff
@@ -0,0 +1,157 @@
+"""
+Support for switch devices that can be controlled using the RaspyRFM rc module.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch/raspyrfm/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PLATFORM, CONF_PORT, CONF_SWITCHES,
+    DEVICE_DEFAULT_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['raspyrfm-client==1.2.8']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_MANUFACTURER = 'gateway_manufacturer'
+CONF_GATEWAY_MODEL = 'gateway_model'
+CONF_CONTROLUNIT_MANUFACTURER = 'controlunit_manufacturer'
+CONF_CONTROLUNIT_MODEL = 'controlunit_model'
+CONF_CHANNEL_CONFIG = 'channel_config'
+DEFAULT_HOST = '127.0.0.1'
+
+# define configuration parameters
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'raspyrfm',
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=None):
+        vol.All(vol.Coerce(int), vol.Range(min=1, max=65535)),
+    vol.Optional(CONF_GATEWAY_MANUFACTURER): cv.string,
+    vol.Optional(CONF_GATEWAY_MODEL): cv.string,
+    vol.Required(CONF_SWITCHES, default=[]): vol.Schema([{
+        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MANUFACTURER): cv.string,
+        vol.Required(CONF_CONTROLUNIT_MODEL): cv.string,
+        vol.Required(CONF_CHANNEL_CONFIG): cv.match_all,
+    }])
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the RaspyRFM switch."""
+    _LOGGER.info("initializing RaspyRFM Switch")
```

Comment:
Remove this. It's already logged by the core.

Reply:
Ok